### PR TITLE
Publish pre-release docs for PRs & main branch

### PR DESCRIPTION
**Pull request by raminqaf**

### CODE REVIEW

**Comment by sujuka99** [comment]
Did you test it somehow?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -79,3 +79,27 @@ jobs:
     uses: bakdata/ci-templates/.github/workflows/python-poetry-publish-snapshot.yaml@v1.25.0
     secrets:
       pypi-token: ${{ secrets.TEST_PYPI_TOKEN }}
+
+  publish-docs-to-dev:
```
Comment (disrupted) [inline .github/workflows/ci.yaml:83]:
```suggestion
  publish-docs-to-dev:
    name: Publish dev docs
    needs: [test]
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -79,3 +79,27 @@ jobs:
...
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3   
+        with:
+          ref: ${{ github.head_ref }}
```
Comment (disrupted) [inline .github/workflows/ci.yaml:88]: why?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (disrupted) [inline .github/workflows/ci.yaml:83]: should not run from main, it should be a separate job

File: .github/actions/update-docs/action.yml [hunk 3]
```diff
@@ -17,6 +17,10 @@ inputs:
   tag:
     description: "Version tag"
     required: false
+  dev:
+    description: "Should push docs to dev (defaults false)"
+    default: "false"
```
Comment (disrupted) [inline .github/actions/update-docs/action.yml:22]: I think a name would be more flexible than a boolean, then we can do `dev` and `main` for example

**Review by disrupted — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (raminqaf) [reply .github/workflows/ci.yaml:90]: @disrupted This is the error I got: https://github.com/bakdata/kpops/actions/runs/5961286480/job/16170206761?pr=339#step:3:170

And here is the suggested solution:
https://github.com/actions/checkout/issues/317

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (raminqaf) [reply .github/workflows/ci.yaml:100]: But this is a separate job 🤔

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -79,3 +79,45 @@ jobs:
...
+  publish-docs-to-main:
+    runs-on: ubuntu-22.04
+    if: ${{ github.head_ref == 'refs/heads/main' }}
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v3   
+        with:
+          ref: ${{ github.head_ref }}
+      
+      - name: Publish docs to dev
```
Comment (disrupted) [inline .github/workflows/ci.yaml:92]: wrong name

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (raminqaf) [reply .github/workflows/ci.yaml:100]: Added the job only on PRs

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -79,3 +79,46 @@ jobs:
...
+  publish-docs-from-main:
+    runs-on: ubuntu-22.04
+    if: ${{ github.head_ref == 'refs/heads/main' }}
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v3   
+        with:
+          ref: ${{ github.head_ref }}
+      
+      - name: Publish docs from
```
Comment (disrupted) [inline .github/workflows/ci.yaml:92]: incomplete name

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -79,3 +79,46 @@ jobs:
     uses: bakdata/ci-templates/.github/workflows/python-poetry-publish-snapshot.yaml@v1.25.0
     secrets:
       pypi-token: ${{ secrets.TEST_PYPI_TOKEN }}
+  
+  publish-docs-from-main:
+    runs-on: ubuntu-22.04
+    if: ${{ github.head_ref == 'refs/heads/main' }}
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v3
```
Comment (disrupted) [inline .github/workflows/ci.yaml:88]: nit: trailing whitespace

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -79,3 +79,46 @@ jobs:
...
+      
+      - name: Publish docs from 
+        uses: ./.github/actions/update-docs
+        with:
+          username: ${{ secrets.GH_USERNAME }}
+          email: ${{ secrets.GH_EMAIL }}
+          token: ${{ secrets.GH_TOKEN }}
+          version: main 
+
+  publish-docs-to-dev:
```
Comment (disrupted) [inline .github/workflows/ci.yaml:100]:
```suggestion
  publish-dev-docs-from-pr:
```

File: .github/workflows/ci.yaml [hunk 8]
```diff
@@ -79,3 +79,46 @@ jobs:
...
+    runs-on: ubuntu-22.04
+    if: ${{ github.event_name == 'pull_request' }}
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v3
```
Comment (disrupted) [inline .github/workflows/ci.yaml:105]: nit: trailing whitespace

File: .github/actions/update-docs/action.yml [hunk 9]
```diff
@@ -27,23 +30,26 @@ runs:
...
+
+        git pull 
+
+    - name: Update documentation dev branch with mike
+      shell: bash
+      if: ${{ inputs.latest == 'false' }}
+      run: |
+        poetry run mike deploy ${{ inputs.version }} --push --rebase --config-file ./docs/mkdocs.yml
+
+
```
Comment (disrupted) [inline .github/actions/update-docs/action.yml:50]: remove extra newline

File: .github/actions/update-docs/action.yml [hunk 10]
```diff
@@ -27,23 +30,26 @@ runs:
...
-          poetry run mike deploy "$new_tag" latest --update-aliases --push --rebase --config-file ./docs/mkdocs.yml
-        fi
+
+        git config --local user.name  ${{ inputs.username }}
+        git config --local user.email ${{ inputs.email }}
+        git config --local user.password ${{ inputs.token }}
+
+        git pull 
+
+    - name: Update documentation dev branch with mike
```
Comment (disrupted) [inline .github/actions/update-docs/action.yml:44]: what is the "documentation dev branch"?

**Review by disrupted — CHANGES_REQUESTED** [review]
please update the PR title

**Comment by raminqaf** [comment]
> please update the PR title

Done

File: .github/actions/update-docs/action.yml [hunk 11]
```diff
@@ -14,8 +13,12 @@ inputs:
   token:
     description: "GitHub Token (must be a PAT for repository dispatch)"
     required: true
-  tag:
-    description: "Version tag"
+  version:
+    description: "Version name to be deployed by mike"
+    required: true 
+  latest:
```
Comment (disrupted) [inline .github/actions/update-docs/action.yml:19]: this is the distinction between release and dev, right? if so I would prefer to call the parameter `release` or `is-release`

File: .github/actions/update-docs/action.yml [hunk 12]
```diff
@@ -27,23 +30,25 @@ runs:
         poetry-version: "1.5.1"
         python-version: "3.10"
 
-    - name: Update documentation branch with mike
+    - name: Install docs dependencies, setup git, and update branch
```
Comment (disrupted) [inline .github/actions/update-docs/action.yml:33]: why not make it multiple steps?

**Review by disrupted — CHANGES_REQUESTED** [review]

File: .github/actions/update-docs/action.yml (hunk 11 above)
Comment (raminqaf) [reply .github/actions/update-docs/action.yml]: done

File: .github/actions/update-docs/action.yml (hunk 12 above)
Comment (raminqaf) [reply .github/actions/update-docs/action.yml]: done

**Review by disrupted — APPROVED** [review]

**Pull request merged**